File: backend/app/models/stats.py
from datetime import datetime

from pydantic import BaseModel

from app.models.common import PyObjectId


class MonthlyStats(BaseModel):
    year: int
    month: int
    books_added_count: int
    reviews_written_count: int
    average_rating: float | None = None


class CalendarDayBookInfo(BaseModel):
    id: PyObjectId
    title: str


class CalendarDayReview(BaseModel):
    id: PyObjectId
    book: CalendarDayBookInfo
    rating: int
    summary_preview: str
    created_at: datetime

File: frontend/src/types/stats.ts
export interface MonthlyStats {
  year: number;
  month: number;
  books_added_count: number;
  reviews_written_count: number;
  average_rating: number | null;
}

/** 날짜 문자열(YYYY-MM-DD) -> 해당 날짜의 리뷰 개수 */
export type CalendarCounts = Record<string, number>;

export interface CalendarDayReview {
  id: string;
  book: {
    id: string;
    title: string;
  };
  rating: number;
  summary_preview: string;
  created_at: string;
}
